tests/integration/test_diagnostic_run.py

```diff
@@ -52,23 +51,22 @@ def check(result_file):
     }
     missing = required_keys - set(result)
     assert not missing
+    unwanted_keys = ['profile_diagnostic', ]
+    for unwanted_key in unwanted_keys:
+        assert unwanted_key not in result
 
 
 SCRIPTS = {
-    # TODO: make independent from ESMValTool installation
-    #     'diagnostic.py':
-    #     dedent("""
-    #         import yaml
-    #         from esmvaltool.diag_scripts.shared import run_diagnostic
-    #
-    #         def main(cfg):
-    #             with open(cfg['setting_name'], 'w') as file:
-    #                 yaml.safe_dump(cfg, file)
-    #
-    #         if __name__ == '__main__':
-    #             with run_diagnostic() as config:
-    #                 main(config)
-    #         """),
+    'diagnostic.py':
+    dedent("""
+        import os
+
+        def main():
+            os.system('cp settings.yml ../../../../../result.yml')
+
+        if __name__ == '__main__':
+            main()
+        """),
```

Comment:
That looks a bit odd. Why is this change needed in this PR and do you check this script somewhere? I would expect something like `assert file_exists('result.yml')` somewhere in the code.

Reply:
it's a hack - it needs that `result.yml` to be checked, without running the actual diagnostic, so am just copying it from where the monkeypatched test ran to where the the file should be in the case of an actual run - maybe @bouweandela can think of a more elegant solution, I didn't want to change too much of his original test, hence the hacky bit here

Reply:
well aren't a big ray of sunshine today :rofl: OK seriously now, I'll look into getting rid of this hack :+1:

Reply:
OK am a total loss here - @bouweandela how do you get a Python diagnostic to write a `results.yml` file with diag env variables like `run_dir` etc without accessing neither the environment (I don't want to add `import os` stuff in the diag) nor by importing `run_diagnostic` from esmvaltool since that package is not supposed to be installed?

Reply:
or @schlunma - don't matter who answers me, as long as we have an answer to fix that ugly hack :grin:

Reply:
Looks like `settings.yml` is located in the cwd, so just load this with `yaml.safe_load()` and then simply write a file to `run_dir` or copy `settings.yml` using `shutil.copy()`.

Reply:
The point I was trying to make in my previous comment is that you do not need to simulate a diagnostic run to test the feature you're implementing in this pull request. Why are you editing this test at all? The only thing you need to test is that the interface file (settings.yml) gets written correctly.

Reply:
no absolutely not, you are right there Bouwe, but we need to run a sim diag to test the tasking mechanism, that mecahnism is not tested otherwise, where do you think that big bump in coverage percentage comes from?